tests/test_cli.py

```diff
@@ -27,3 +29,90 @@ def test_cli_should_return_help_string_when_passed_no_args():
     # THEN
     assert result.returncode == 0
     assert result.stdout.strip() == expected.strip()
+
+@patch("edgar_tool.text_search.write_results_to_file")
```

Comment:
What would you think about using pytest's [tmp_path](https://docs.pytest.org/en/stable/how-to/tmp_path.html#the-tmp-path-fixture) fixture? It was made for this exact thing! 🪄 

```python
def test_example(tmp_path):
    # GIVEN
    output_file = tmp_path / "results.csv"

    # WHEN
    SecEdgarScraperCli.text_search(
        "John Doe",
        output=str(output_file),
        start_date="2021-01-01",
        end_date="2021-01-31"
    )

    # THEN
    written_data = output_file.read_text()
    # And you can make your assertions directly against the content
    #  of the file - it's truly end-to-end
    assert "'root_form', 'form_name'" in written_data
```